### PR TITLE
added test with debug mode

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
When running on CortexXSOAR 6.0.0+ it'll use the debug test module instead of the basic one.

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

### CODE REVIEW

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py [hunk 1]
```diff
@@ -36,11 +36,12 @@ def save_configuration(arguments: dict) -> List[str]:
     return get_errors(res)
 
 
-def execute_test_module(arguments: dict) -> List[str]:
+def execute_test_module_legacy(arguments: dict) -> List[str]:
```
Comment (Itay4) [inline Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:39]: i think its a confusing name as this is no legacy functionality, infact it is the standard test module
i would keep the name for this and rename the new func to mention it is `debug`

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py [hunk 2]
```diff
@@ -74,11 +93,15 @@ def main():
             }
         }
         if errors:
-            err_str = "\n".join(errors)
+            err_str = '\n'.join(errors[0].splitlines()[:10])
```
Comment (Itay4) [inline Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:96]: what are we doing here? why are we displaying the first lines of the error?

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py [hunk 3]
```diff
@@ -74,11 +93,15 @@ def main():
...
             human_readable = f'Found errors in instance {instance_name} after changing the next keys: ' \
                              f'{", ".join(keys_in_instance)}\n' \
-                             f'Errors:\n{err_str}'
-            return_error(human_readable, outputs=context)
+                             f'First lines of the error:\n{err_str}'
+            demisto.results(fileResult(
+                'test-module-results.txt',
```
Comment (Itay4) [inline Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:101]: can we add a test for this?

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py (hunk 2 above)
Comment (jochman) [reply Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:96]: exactly

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py (hunk 3 above)
Comment (jochman) [reply Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:101]: for what?

**Comment by jochman** [comment]
@Itay4 ?

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py (hunk 2 above)
Comment (Itay4) [reply Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:97]: exactly what? why are we doing this?

File: Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py (hunk 3 above)
Comment (Itay4) [reply Packs/Troubleshoot/Scripts/TroubleshootTestInstance/TroubleshootTestInstance.py:102]: to see we get the result file as expected

**Pull request closed without merging**